src/routes/settings/profile/+page.svelte

```diff
@@ -15,31 +16,45 @@
 	};
 	let last_contact = default_contact;
 
+	let updating = false;
 	async function update_profile() {
-		let encodedAvatar = "";
-		if (avatar_files !== null && avatar_files.length > 0) {
-			console.log("base64 encode avatar file");
-			const bytes = new Uint8Array(await avatar_files[0].arrayBuffer());
-			let byteString = "";
-			for (const b of bytes) {
-				byteString += String.fromCharCode(b);
-			}
-			encodedAvatar = window.btoa(byteString);
+		if (updating) {
+			return;
 		}
+		updating = true;
+
+		try {
+			let encodedAvatar = "";
```

Comment:
I forgot that I should name it `encoded_avatar`, also, `byte_string` below.

Reply:
Let's fine, I believe it'll be fix in the future refactoring (if any) about this part.